src/jit/lsra.cpp

```diff
@@ -6024,7 +6309,55 @@ regNumber LinearScan::allocateBusyReg(Interval* current, RefPosition* refPositio
     if (farthestRefPhysRegRecord != nullptr)
     {
         foundReg = farthestRefPhysRegRecord->regNum;
+#ifdef _TARGET_ARM_
+        if (genIsValidDoubleReg(foundReg))
+        {
+            // For a double register, we has following three cases.
+            // Case 1: farthestRefPhysRegRecord is assigned to TYP_DOUBLE interval
+            // Case 2: farthestRefPhysRegRecord and farthestRefPhysRegRecord2 are assigned to
+            //         different TYP_FLOAT intervals
+            // Case 3: farthestRefPhysRegRecord is assgined to TYP_FLOAT interval
+            //         and farthestRefPhysRegRecord2 is nullptr
+            // Case 4: farthestRefPhysRegRecord is nullptr, and farthestRefPhysRegRecord2 is
+            //         assigned to a TYP_FLOAT interval
+            if (farthestRefPhysRegRecord->assignedInterval != nullptr)
+            {
+                if (farthestRefPhysRegRecord->assignedInterval->registerType == TYP_DOUBLE)
+                {
+                    // Case 1: farthestRefPhysRegRecord is assigned to TYP_DOUBLE interval
+                    unassignPhysReg(farthestRefPhysRegRecord,
+                                    farthestRefPhysRegRecord->assignedInterval->recentRefPosition);
+                }
+                else
+                {
+                    // Case 2: farthestRefPhysRegRecord and farthestRefPhysRegRecord2 are assigned to
+                    //         different TYP_FLOAT intervals
+                    // Case 3: farthestRefPhysRegRecord is assgined to TYP_FLOAT interval
+                    //         and farthestRefPhysRegRecord2 is nullptr
+                    unassignPhysReg(farthestRefPhysRegRecord,
+                                    farthestRefPhysRegRecord->assignedInterval->recentRefPosition);
+                    if (farthestRefPhysRegRecord2 != nullptr)
+                        unassignPhysReg(farthestRefPhysRegRecord2,
+                                        farthestRefPhysRegRecord2->assignedInterval->recentRefPosition);
+                }
+            }
+            else
+            {
+                // Case 4: farthestRefPhysRegRecord is nullptr, and farthestRefPhysRegRecord2 is
+                //         assigned to a TYP_FLOAT interval
+                assert(farthestRefPhysRegRecord2->assignedInterval != nullptr);
+                assert(farthestRefPhysRegRecord2->assignedInterval->registerType == TYP_FLOAT);
+                unassignPhysReg(farthestRefPhysRegRecord2,
+                                farthestRefPhysRegRecord2->assignedInterval->recentRefPosition);
+            }
+        }
+        else
+        {
+            unassignPhysReg(farthestRefPhysRegRecord, farthestRefPhysRegRecord->assignedInterval->recentRefPosition);
+        }
+#else
         unassignPhysReg(farthestRefPhysRegRecord, farthestRefPhysRegRecord->assignedInterval->recentRefPosition);
```

Comment:
You could keep the above call to `unassignPhysReg` in common, and avoid repeating the non-double case. Also, I think it's worth extracting the double case into a method `unassignDoublePhysReg(RegRecord* doubleRegRecord)` that always spills the active `assignedInterval`(s) at their `recentRefPosition`.